src/image_surface.rs

```diff
@@ -44,6 +110,31 @@ impl ImageSurface {
         }
     }
 
+    #[cfg(feature = "png-functions")]
+    pub fn create_from_png_stream<R: Read>(stream: R) -> io::Result<ImageSurface> {
```

Comment:
I believe the best approximation of [this guideline](http://cairographics.org/manual/bindings-overloading.html) would be to avoid the `_stream` suffix as this is the more generic version. Optionally there could be variants that take `P: AsRef<Path>`.